src/process/run.cr

```diff
@@ -93,11 +106,21 @@ class Process
         reopen_io(fork_output || output, STDOUT, "w")
         reopen_io(fork_error || error, STDERR, "w")
 
-        # Dir.chdir(chdir) if chdir
+        ENV.clear if unsetenv_others
+        if env
+          env.not_nil!.each do |key, val|
```

Comment:
Did you test if the `.not_nil!` is still needed after the `env = Env` bug was fixed? I doubt it is.